ecs-agent/gmsacredclient/credentialsfetcherclient_test.go

```diff
@@ -133,6 +206,128 @@ func TestCredentialsFetcherClient_AddKerberosLease(t *testing.T) {
 	}
 }
 
+func TestCredentialsFetcherClient_AddNonDomainJoinedKerberosArnLease(t *testing.T) {
+	tests := []struct {
+		name          string
+		credspecArn   []string
+		accessId      string
+		secretKey     string
+		sessionToken  string
+		region        string
+		response      CredentialsFetcherArnResponse
+		expectedError string
+	}{
+		{
+			"invalid request empty credspecArn contents",
+			[]string{},
+			"testusername",
+			"testpassword",
+			"testdomain",
+			"testregion",
+			CredentialsFetcherArnResponse{},
+			"rpc error: code = InvalidArgument desc = credentialspecArns should not be empty",
+		},
+		{
+			"invalid request username, password or domain should not be empty",
+			[]string{credspec_webapp01},
+			"",
+			"",
+			"",
+			"",
+			CredentialsFetcherArnResponse{},
+			"rpc error: code = InvalidArgument desc = accessid, secretkey, sessiontoken or region should not be empty",
+		},
+		{
+			"valid request credspecs associated to gMSA account",
+			[]string{credspec_webapp01},
+			"testusername",
+			"testpassword",
+			"testdomain",
+			"testregion",
+			CredentialsFetcherArnResponse{LeaseID: "123456", KerberosTicketsMap: map[string]string{"arn:aws:s3:::gmsacredspec/gmsa-cred-spec.json": "/var/credentials-fetcher/krbdir/123456/webapp01"}},
+			"",
+		},
+	}
+
+	ctx := context.Background()
+
+	conn, err := grpc.DialContext(ctx, "", grpc.WithInsecure(), grpc.WithContextDialer(dialer()))
```

Comment:
Does this make an actual connection to a server? If so, how is that server set up?

Reply:
test client communicates with the mock server